Index: petsc/config/BuildSystem/config/package.py
===================================================================
--- petsc.orig/config/BuildSystem/config/package.py	2024-05-16 13:03:24.428845150 +0200
+++ petsc/config/BuildSystem/config/package.py	2024-05-16 13:03:24.428845150 +0200
@@ -989,7 +989,7 @@
       if not hasattr(package, 'found'):
         raise RuntimeError('Package '+package.name+' does not have found attribute!')
       if not package.found:
-        if self.argDB['with-'+package.package] == 1:
+        if 'with-'+package.package in self.framework.clArgDB and self.framework.clArgDB['with-'+package.package] == 1:
           raise RuntimeError('Package '+package.PACKAGE+' needed by '+self.name+' failed to configure.\nMail configure.log to petsc-maint@mcs.anl.gov.')
         else:
           str = ''
