Small adaptations to use Debian sprng.

--- a/src/sys/classes/random/impls/sprng/sprng.c
+++ b/src/sys/classes/random/impls/sprng/sprng.c
@@ -9,7 +9,7 @@
 static PetscErrorCode PetscRandomSeed_Sprng(PetscRandom r)
 {
   PetscFunctionBegin;
-  init_sprng(r->seed, SPRNG_DEFAULT);
+  init_sprng(SPRNG_CMRG,r->seed, SPRNG_DEFAULT);
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
--- a/config/BuildSystem/config/packages/sprng.py
+++ b/config/BuildSystem/config/packages/sprng.py
@@ -6,7 +6,7 @@
     self.download  = ['https://web.cels.anl.gov/projects/petsc/download/externalpackages/sprng-1.0.tar.gz']
     self.functions = ['make_new_seed_mpi']
     self.includes  = ['sprng.h']
-    self.liblist   = [['liblcg.a']]
+    self.liblist   = [['libsprng.a']]
     return
 
   def setupDependencies(self, framework):
